src/NetTopologySuite/IO/WKTReader.cs

```diff
@@ -104,17 +98,16 @@ public bool IsOldNtsMultiPointSyntaxAllowed
         /// </summary>
         public GeometryFactory Factory
         {
-            get => _ntsGeometryServices.CreateGeometryFactory(_precisionModel, DefaultSRID, _coordinateSequencefactory);
+            get => _ntsGeometryServices.CreateGeometryFactory(DefaultSRID);
             set
             {
                 if (value != null)
                 {
-                    if (value.GeometryOverlay != _ntsGeometryServices.GeometryOverlay)
-                        throw new ArgumentException("Factory has incompatible set of overlay functions assigned.", nameof(value));
-
-                    _coordinateSequencefactory = value.CoordinateSequenceFactory;
-                    _precisionModel = value.PrecisionModel;
-                    DefaultSRID = value.SRID;
+                    _ntsGeometryServices = (NtsGeometryServices)Activator.CreateInstance(_ntsGeometryServices.GetType(),
```

Comment:
reflection is dangerous; we can't force external subclasses to have a constructor that looks like this, we can't advertise that requirement strongly enough (people won't read the documentation, and only this very unlikely use case would care, so it would go unnoticed until a random time in the future), and it's not really **that** critical to our operation, right?

why not do the same that the `GeometryFactory` constructor does, and mark this property `[Obsolete]`?